stdlib/public/Platform/CMakeLists.txt

```diff
@@ -84,7 +84,19 @@ foreach(sdk ${SWIFT_SDKS})
 
     # Some SDKs place their headers in architecture-specific subfolders.
     if((${sdk} STREQUAL "LINUX" OR ${sdk} STREQUAL "FREEBSD") AND CMAKE_LIBRARY_ARCHITECTURE)
-      set(GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH "${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+      # Reviving this comment from earlier commit:
+      # FIXME: Some distributions install headers in
+      #        "/usr/include/x86_64-linux-gnu/sys/...". Here we use the host
+      #        machine's path, regardless of the SDK target we're building for.
+      #        This will break if cross-compiling from a distro that uses the
+      #        architecture as part of the path to a distro that does not.
+      #
+      #        However, this is not always the case. As a stopgap solution, we
+      #        will set the GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH depending
+      #        on if the arch_include_path/library_arch path exists.
+      if(EXISTS "${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+        set(GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH "${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}")
+      endif()
```

Comment:
If I'm seeing this correctly, `GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH` ends up being passed to two GYB invocations below. If this value is not set, those invocations will have an empty string for that value and the modulemap will be incorrect. Or am I missing something?

I might be wrong, but I think `CMAKE_LIBRARY_ARCHITECTURE` is empty for systems that do not use it, so the previous version is basically like setting `GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH` to the previous value with and extra trailing separator.

Reply:
@drodriguez For runtimes libcxx builds it gets set to the runtime target name.

This is a check to see if the system has a directory like /usr/include/linux-x86_64/ (ie ${GLIBC_SYSROOT_RELATIVE_ARCH_INCLUDE_PATH}/${CMAKE_LIBRARY_ARCHITECTURE}). Exists or not, and if not then do not append CMAKE_LIBRARY_ARCHITECTURE to the path. 

In this case the CMAKE_LIBRARY_ARCHITECTURE is not empty; it's set to the target name for the runtime.

Reply:
I will try and set the CMAKE_LIBRARY_ARCHITECTURE to empty in cmake cache files and see what happens. I still think the comment should probably remain.

Reply:
@drodriguez Still trying to understand how swift stdlib is generating the CMakeCCompiler.cmake/CMakeCXXCompiler.cmake files that appear to set CMAKE_LIBRARY_ARCHITECTURE.

Reply:
I wonder if this is something with our CentOS. Or did you experience this in other OS? Maybe tomorrow we can check this in a Ubuntu machine. I believe what you are telling me, and if the value of `CMAKE_LIBRARY_ARCHITECTURE` is not empty, the fix is correct, but I wonder what's happening here.

Reply:
@drodriguez @compnerd I've updated this patch to just provide a way to have an override path to GLIBC_SYSROOT_RELATIVE.*_INCLUDE_PATH. What do you all think?